### PR TITLE
Remove link to 2FA push method and don't stop polling

### PR DESCRIPTION
**Pull request by yurynix**
This PR:
* Removes the stopping of polling from push auth to allow user approving the push even if she navigated away ( like the sms screen )
* Removes the link to the push screen as we currently not re-sending the push notification.

Another take on #14615

#### Testing instructions
1. Run `git checkout update/update/2fa-remove-push-link` and start your server, or open a [live branch](https://calypso.live/?branch=update/2fa-remove-push-link)
2. Open the [`Login` page](http://calypso.localhost:3000/log-in)
3. Login with account that has 2FA with push, assert you presented with the push polling screen.
4. Click `Code via text message`
5. Assert there is no link to `push` - `Login with WordPress.com app`
6. Assert you received SMS message with a code, however click `approve` on the push notification instead.
7. Assert you're indeed logged-in.

#### Reviews
  
- [x] Code
- [x] Product

### CODE REVIEW

**Comment by drewblaisdell** [comment]
What do you think about pulling in the poller component from [this PR](https://github.com/Automattic/wp-calypso/pull/14615)? It will cause polling to stop when the user navigates out of the 2FA parts of `/log-in`.

**Comment by drewblaisdell** [comment]
I updated this such to add the aforementioned polling component. If you're cool with that, this LGTM. I saved your first commit on Trello, but this seems like a better approach to me than keeping a reference to a callback on an action object.

**Comment by stephanethomas** [comment]
The code looks great and works fine.

This poller component is an awesome idea @drewblaisdell, it's super neat.

**Comment by fabianapsimoes** [comment]
Product :+1:

**Comment by gziolo** [comment]
screen shot 2017-06-02 at 12 07 24

Steps to reproduce:

* Try to log in with an account with sms set as verification method
* Wait for push notification screen
* Click use text message instead
* You should see missing link after `Continue with` header

File: client/blocks/login/index.jsx [hunk 1]
```diff
@@ -108,25 +109,37 @@ class Login extends Component {
...
+
 		if ( twoFactorEnabled && includes( [ 'authenticator', 'sms', 'backup' ], twoFactorAuthType ) ) {
 			return (
-				<VerificationCodeForm
-					rememberMe={ rememberMe }
-					onSuccess={ this.rebootAfterLogin }
-					twoFactorAuthType={ twoFactorAuthType }
-				/>
+				<div>
+					{ poller }
```
Comment (gziolo) [inline client/blocks/login/index.jsx]: It doesn't get unmounted when navigating to a different 2FA screen. Pretty cool!

**Comment by yurynix** [comment]
@drewblaisdell Thank you for updating this PR 👍

**Pull request merged**